crates/fluvio/src/sync/mod.rs

```diff
@@ -108,17 +108,19 @@ mod context {
             use tokio::select;
             use fluvio_future::timer::sleep;
 
+            // We can short circuit here if already present
+            let found = search(self.store().read().await);
+            if found.is_some() {
+                return Ok(found);
+            }
```

Comment:
Good catch.   Let's tighten code here like:
```
if let Some(found) = search(self.store().read().await) {
     return found
}
```

Surprised clippy didn't catch that

Reply:
Clippy didn't catch because it tightens to
```
if let Some(found) = search(self.store().read().await) {
     return Ok(Some(found))
}
```
I think that's still better. I'll change it.